.github/workflows/eden.yml

```diff
@@ -17,24 +17,63 @@ on:
       - 'docs/**'
       - 'pkg/pillar/docs/**'
 
+
 concurrency:
   group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
   cancel-in-progress: true
 
 jobs:
+  trigger_pr_publish:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Trigger PR Publish for Eden
+        run: |
+          pr_build_job=$(curl -s -X POST -H "Authorization: Bearer ${{ secrets.GITHUB_TOKEN }}" \
+            -H "Accept: application/vnd.github.v3+json" \
+            https://api.github.com/repos/${{ github.repository }}/actions/workflows/danger.yml/dispatches \
+            -d '{"ref": "${{ github.head_ref }}"}')
+  
+          echo "PR Build triggered with response: $pr_build_job"
+          job_id=$(echo "$response" | jq -r '.job_id')
+
+          status="unknown"
+          max_attempts=10
+          current_attempt=0
+
+          while [ "$status" != "success" ] && [ "$status" != "failure" ] && [ $current_attempt -lt $max_attempts ]; do
+            job_status=$(curl -s -H "Authorization: Bearer ${{ secrets.GITHUB_TOKEN }}" -H "Accept: application/vnd.github.v3+json" \
+                "https://api.github.com/repos/${{ github.repository }}/actions/jobs/$job_id" | jq -r '.conclusion')
+
+            echo "Job status: $job_status"
+
+            if [ "$job_status" != "success" ] && [ "$job_status" != "failure" ]; then
+              echo "Job is still running. Checking again in 60 seconds."
+              sleep 60
+            else
+              status="$job_status"
+            fi
+
+            current_attempt=$((current_attempt + 1))
+          done
+
+          echo "Final job status: $status"
```

Comment:
I think we can save the result of the job to env and check it in the later steps.
```
echo "PUSH_RESULT=$status" >> $GITHUB_ENV
```
and use it later with `env.PUSH_RESULT`

Reply:
What if your jobs are running on different machines? I'm not sure GHA will copy env variables between workflows. (It's shared between jobs inside workflows though) And if you are using env variable to wait for it, you have to do it inside the job, meaning runner will be running, which will slow down whole pipeline (because we have limited pull of runners)